internal/provider/data_source_gitlab_branch.go

```diff
@@ -0,0 +1,101 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+var _ = registerDataSource("gitlab_branch", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "Provide details about a gitlab project branch \n\n" +
+			"> **Note**: name and project must be specified.",
```

Comment:
I think we can avoid that comment. The `required` arguments are documented in a well-known format already :)

Reply:
looks resolved to me